src/fe-gtk/maingui.c

```diff
@@ -1084,13 +1084,39 @@ mg_tab_close_cb (GtkWidget *dialog, gint arg1, session *sess)
 	}
 }
 
+typedef struct closed_channel {
+	server *server;
+	char channel[CHANLEN];
+	char key[64];
+} closed_channel;
+
+static GQueue closed_channels = G_QUEUE_INIT;
+
 void
 mg_tab_close (session *sess)
 {
 	GtkWidget *dialog;
 	GSList *list;
 	int i;
 
+	if (hexchat_is_quitting)
+	{
+		closed_channel *chan;
+		while (!g_queue_is_empty(&closed_channels))
```

Comment:
About g_queue_free_full, isn't it intended to be used only for dynamically allocated queue objects? I thought it was the same as g_queue_free, so I avoided using it. I need to investigate a bit further.